azext_iot/common/certops.py

```diff
@@ -125,3 +134,198 @@ def open_certificate(certificate_path: str) -> str:
         raise ValueError("Certificate file type must be either '.pem' or '.cer'.")
     # Remove trailing white space from the certificate content
     return certificate.rstrip()
+
+
+# TODO - Unit test, compare with test_utils::_generate_root_certificate
+def create_root_certificate(
+    subject: Optional[str] = "Azure_IoT_CLI_Extension_Cert",
+    valid_days: Optional[int] = 365,
```

Comment:
Is `Optional` appropriate here? What does `valid_days = None` mean?

Reply:
Like you've probably seen elsewhere in my usage of `Optional`, this is likely my Typescript brain fighting with my Python brain :)

I'll make another pass at updating method signatures and object/tuple definitions.